src/server/status/stats.js

```diff
@@ -0,0 +1,12 @@
+import { SavedObjectsClient } from '../saved_objects/client';
+
+export async function getStats(kibanaIndex, request, callWithRequest) {
+  const savedObjectsClient = new SavedObjectsClient(kibanaIndex, request, callWithRequest);
+  const dashResponse = await savedObjectsClient.find({ type: 'dashboard', perPage: 0 });
+  const visResponse = await savedObjectsClient.find({ type: 'visualization', perPage: 0 });
```

Comment:
These stats might not need to be perfectly real-time. Feel free to do whatever you like with this comment, but one idea could be to create an internal poller that caches the stats every 10 minutes or so, and then when the API is called, it just returned the last stats from the cache.